modules/oe_content_featured_media_field/src/Plugin/Field/FieldType/FeaturedMediaFieldItem.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_content_featured_media_field\Plugin\Field\FieldType;
+
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+use Drupal\Core\Field\Plugin\Field\FieldType\EntityReferenceItem;
+use Drupal\Core\Field\PreconfiguredFieldUiOptionsInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\TypedData\DataDefinition;
+use Drupal\Core\TypedData\OptionsProviderInterface;
+
+/**
+ * Plugin implementation of the 'oe_featured_media' field type.
+ *
+ * @FieldType(
+ *   id = "oe_featured_media",
+ *   label = @Translation("Featured media"),
+ *   module = "oe_content_featured_media_field",
+ *   category = @Translation("OpenEuropa"),
+ *   description = @Translation("Stores a featured media item and caption."),
+ *   default_formatter = "oe_featured_media_label",
+ *   default_widget = "oe_featured_media_widget",
+ *   column_groups = {
+ *     "target_id" = {
+ *       "label" = @Translation("Media item"),
+ *       "translatable" = FALSE
+ *     },
+ *     "caption" = {
+ *       "label" = @Translation("Caption"),
+ *       "translatable" = TRUE
+ *     },
+ *   },
+ *   list_class = "\Drupal\Core\Field\EntityReferenceFieldItemList"
+ * )
+ */
+class FeaturedMediaFieldItem extends EntityReferenceItem implements OptionsProviderInterface, PreconfiguredFieldUiOptionsInterface {
+
+  /**
```

Comment:
I can fill in Caption and save the node and when I edit it, my value is not saved.
We might need to add states if caption is filled in media field becomes required or a constraint.

Reply:
I will add states for this.